From 0109d1be617359947ae4e9746b1284acc5a4d526 Mon Sep 17 00:00:00 2001
From: Umashankar Godachi <umasha@codeaurora.org>
Date: Mon, 8 Oct 2018 15:57:47 +0530
Subject: [PATCH 1/4] Fix to release partial wakelock for SRVCC handover.

During SRVCC handover partial wakelock is acquired while
creating new GsmCdmaConnection, but is not released till
the call is disconnected, leading to Application processor
not going to sleep.

Fix: Release wakelock considering the ACTIVE,HOLDING pre
     -handover states.

Change-Id: I650a1a9bb7df16ebf2e6803635d38bad5db7c56c
CRs-Fixed: 2328958
Signed-off-by: DennySPB <dennyspb@gmail.com>
Signed-off-by: Kazakov Stepan <ksrt12group@gmail.com>
---
 .../com/android/internal/telephony/GsmCdmaCallTracker.java     | 3 +++
 src/java/com/android/internal/telephony/GsmCdmaConnection.java | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)
 mode change 100755 => 100644 src/java/com/android/internal/telephony/GsmCdmaCallTracker.java

diff --git a/src/java/com/android/internal/telephony/GsmCdmaCallTracker.java b/src/java/com/android/internal/telephony/GsmCdmaCallTracker.java
old mode 100755
new mode 100644
index 1b43167cf..9d389bd8d
--- a/src/java/com/android/internal/telephony/GsmCdmaCallTracker.java
+++ b/src/java/com/android/internal/telephony/GsmCdmaCallTracker.java
@@ -843,6 +843,9 @@ public class GsmCdmaCallTracker extends CallTracker {
                                 hoConnection.mPreHandoverState != GsmCdmaCall.State.HOLDING &&
                                 dc.state == DriverCall.State.ACTIVE) {
                             mConnections[i].onConnectedInOrOut();
+                        } else if (dc.state == DriverCall.State.ACTIVE
+                                || dc.state == DriverCall.State.HOLDING) {
+                            mConnections[i].releaseWakeLock();
                         }
 
                         mHandoverConnections.remove(hoConnection);
diff --git a/src/java/com/android/internal/telephony/GsmCdmaConnection.java b/src/java/com/android/internal/telephony/GsmCdmaConnection.java
index 603a0b70b..7bddb7554 100644
--- a/src/java/com/android/internal/telephony/GsmCdmaConnection.java
+++ b/src/java/com/android/internal/telephony/GsmCdmaConnection.java
@@ -1121,7 +1121,7 @@ public class GsmCdmaConnection extends Connection {
         }
     }
 
-    private void releaseWakeLock() {
+    void releaseWakeLock() {
         if (mPartialWakeLock != null) {
             synchronized (mPartialWakeLock) {
                 if (mPartialWakeLock.isHeld()) {
-- 
2.19.1

